File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/ports/PaymentProvider.java
package it.giannibombelli.wsc2026.payment.domain.ports;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;

import java.util.UUID;

public interface PaymentProvider {
    PaymentProviderResult process(UUID paymentId, UUID providerReference, Money amount);

    PaymentProviderResult refund(UUID paymentId, UUID providerReference, Money amount);
}

File: typescript/src/payment/domain/ports/paymentProvider.ts
// Porta per i provider di pagamento esterni.

import { Uuid } from '@/common/domain/primitive/uuid.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { PaymentProviderResult } from './paymentProviderResult.js';

export interface PaymentProvider {
  process(paymentId: Uuid, providerReference: Uuid, amount: Money): PaymentProviderResult;

  refund(paymentId: Uuid, providerReference: Uuid, amount: Money): PaymentProviderResult;
}
